File: trunplay-backend/src/api/system.py
"""
System API endpoints.
"""
import os
import time
from fastapi import APIRouter

from ..database.models import DB_PATH
from ..database.schemas import PlaybackStatus
from ..services.playback import get_playback_service

router = APIRouter(prefix="/system", tags=["system"])


@router.get("/status")
def get_system_status():
    """Get system status."""
    # Get database size
    db_size = 0
    if os.path.exists(DB_PATH):
        db_size = os.path.getsize(DB_PATH)

    # Get playback status
    playback_service = get_playback_service()
    info = playback_service.get_playback_info()

    return {
        "version": "1.0.0",
        "uptime": int(time.time()),
        "db_path": DB_PATH,
        "db_size": db_size,
        "playback_status": info.status.value,
        "local_ip": get_playback_service()._media_server.get_local_ip()
    }


@router.get("/health")
def health_check():
    """Health check endpoint."""
    return {"status": "ok"}


@router.get("/config")
def get_config():
    """Get system configuration."""
    return {
        "api_port": 8088,
        "media_port": 8089,
        "db_path": DB_PATH,
        "local_media_paths": ["/mnt", "/tmp"],
        "log_level": "INFO"
    }

File: trunplay-backend/internal/api/system.go
package api

import (
	"net/http"
	"os"
	"time"

	"github.com/gin-gonic/gin"
	"github.com/trunplay/trunplay-backend/internal/db"
)

func registerSystem(r *gin.RouterGroup) {
	g := r.Group("/system")
	g.GET("/status", systemStatus)
	g.GET("/health", systemHealth)
	g.GET("/config", systemConfig)
}

func systemStatus(c *gin.Context) {
	dbSize := int64(0)
	if fi, err := os.Stat(db.DBPath); err == nil {
		dbSize = fi.Size()
	}
	info := getPlayback(c).GetPlaybackInfo()
	localIP := getMediaServer(c).GetLocalIP()
	database := getDB(c)
	tx, err := database.Begin()
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"version": "1.0.0", "uptime": time.Now().Unix(), "db_path": db.DBPath, "db_size": dbSize, "playback_status": string(info.Status), "local_ip": localIP, "active_plans": 0, "total_devices": 0})
		return
	}
	defer tx.Rollback()
	plans, _ := db.GetActivePlans(tx)
	devices, _ := db.GetDevices(tx)
	activePlans, totalDevices := 0, 0
	if plans != nil {
		activePlans = len(plans)
	}
	if devices != nil {
		totalDevices = len(devices)
	}
	c.JSON(http.StatusOK, gin.H{
		"version":         "1.0.0",
		"uptime":          time.Now().Unix(),
		"db_path":         db.DBPath,
		"db_size":         dbSize,
		"playback_status": string(info.Status),
		"local_ip":        localIP,
		"active_plans":    activePlans,
		"total_devices":   totalDevices,
	})
}

func systemHealth(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{"status": "ok"})
}

func systemConfig(c *gin.Context) {
	cfg := getConfig(c)
	c.JSON(http.StatusOK, gin.H{
		"api_port":          cfg.APIPort,
		"media_port":        cfg.MediaPort,
		"db_path":           cfg.DBPath,
		"local_media_paths": cfg.LocalMediaPaths,
		"log_level":         cfg.LogLevel,
	})
}
